File: frontend/src/services/courseDesign.ts
import { request } from './api'

/** 课程设计：大纲中的小节草案。 */
export type OutlineSectionDraft = {
  title: string
}

/** 课程设计：大纲中的章节草案。 */
export type OutlineChapterDraft = {
  title: string
  sections: OutlineSectionDraft[]
}

export type GenerateOutlineDraftResp = {
  chapters: OutlineChapterDraft[]
}

export type ApplyOutlineDraftResp = {
  created_chapters: number
  created_sections: number
}

export type GenerateOutlineDraftBody = {
  qa_model_id?: number
  extra_hint?: string
}

/** 生成 AI 课程大纲草案。 */
export async function generateOutlineDraft(token: string, courseId: number, body: GenerateOutlineDraftBody = {}) {
  return request<GenerateOutlineDraftResp>(`/teacher/courses/${courseId}/design/outline-draft/generate`, {
    method: 'POST',
    token,
    body: Object.keys(body).length ? body : {},
  })
}

/** 将编辑后的草案章/节追加写入课程。 */
export async function applyOutlineDraft(token: string, courseId: number, chapters: OutlineChapterDraft[]) {
  return request<ApplyOutlineDraftResp>(`/teacher/courses/${courseId}/design/outline-draft/apply`, {
    method: 'POST',
    token,
    body: { chapters },
  })
}

File: internal/dto/course/course_design.go
package course

// GenerateOutlineDraftReq 请求 AI 根据课程信息生成章节/节大纲草案（仅生成 JSON，不落库）。
type GenerateOutlineDraftReq struct {
	// QAModelID 指定教师名下问答模型 id；省略则使用列表中第一个 qa 模型。
	QAModelID *int64 `json:"qa_model_id"`
	// ExtraHint 教师补充说明（先修要求、侧重方向、周课时等），可为空。
	ExtraHint string `json:"extra_hint"`
}

// OutlineSectionDraft 大纲中的「节」。
type OutlineSectionDraft struct {
	Title string `json:"title"`
}

// OutlineChapterDraft 大纲中的「章」及其下属节。
type OutlineChapterDraft struct {
	Title    string                `json:"title"`
	Sections []OutlineSectionDraft `json:"sections"`
}

// GenerateOutlineDraftResp AI 生成的大纲草案，供前端预览与二次编辑后再应用。
type GenerateOutlineDraftResp struct {
	Chapters []OutlineChapterDraft `json:"chapters"`
}

// ApplyOutlineDraftReq 将教师确认后的大纲写入课程：在现有章节之后追加新章与节。
type ApplyOutlineDraftReq struct {
	Chapters []OutlineChapterDraft `json:"chapters" binding:"required"`
}

// ApplyOutlineDraftResp 应用大纲后的创建数量统计。
type ApplyOutlineDraftResp struct {
	CreatedChapters int `json:"created_chapters"`
	CreatedSections int `json:"created_sections"`
}
